app/serializers/invite_serializer.rb

```diff
@@ -1,21 +1,39 @@
 # frozen_string_literal: true
 
 class InviteSerializer < ApplicationSerializer
-
-  attributes :email, :updated_at, :redeemed_at, :expired, :user
+  attributes :id, :email, :updated_at, :redeemed_at, :expired, :user, :invite_source
 
   def include_email?
-    options[:show_emails] && !object.redeemed?
+    options[:type] == "pending" && options[:show_emails] && !object.redeemed?
+  end
+
+  def redeemed_at
+    options[:type] == "pending" ? nil : object.redeemed_at
   end
 
   def expired
-    object.expired?
+    options[:type] == "pending" ? object.expired? : nil
+  end
+
+  def include_user?
+    options[:type] == "redeemed"
   end
 
   def user
     ser = InvitedUserSerializer.new(object.user, scope: scope, root: false)
-    ser.invited_by = object.invited_by
+    ser.invited_by = object.invite.invited_by
     ser.as_json
   end
 
+  def include_invite_source?
+    options[:type] == "redeemed"
+  end
+
+  def invite_source
+    if object.invite.is_invite_link?
+      "<a href='#{Discourse.base_url}/invites/#{object.invite.invite_key}' target='_blank'>#{object.invite.invite_key[0..5]}</a>"
```

Comment:
Hmm crafting HTML in a serializer on the server side is a red flag to me. Can we do this on the client side instead?